### PR TITLE
Export feature

### PR DESCRIPTION
**Pull request by michaelcullum**
Old pull request: #260 
Issue: #257 

Still needs an icon for the menu at some point

@chartjes Fixed tests on the old (merged then reverted) PR

### CODE REVIEW

**Comment by mdwheele** [comment]
@michaelcullum Thanks much for sending this again. I'm about to pull it down to verify everything works. Otherwise, looks good to me. 

I would like to adjust the route registration to be mounted under `admin/export` to give us a easier path when we flesh out this feature-set. I will make these changes as part of the merge process if you don't see any problem doing so. 

In this specific case, the full routes would end up something like:

```php
$web->get('/admin/export/csv', 'OpenCFP\Http\Controller\Admin\ExportsController::attributedTalksExportAction')->bind('admin_export_csv');
$web->get('/admin/export/csv/anon', 'OpenCFP\Http\Controller\Admin\ExportsController::anonymousTalksExportAction')->bind('admin_export_csv_anon');
$web->get('/admin/export/csv/selected', 'OpenCFP\Http\Controller\Admin\ExportsController::selectedTalksExportAction')->bind('admin_export_csv_selected');
$web->get('/admin/export/csv/emails', 'OpenCFP\Http\Controller\Admin\ExportsController::emailExportAction')->bind('admin_export_csv_emails');
```

I'll let this hang for a day or so after validating 1c30637. Also gives me time to write a few basic tests. Thanks!

File: classes/Domain/Entity/Mapper/Talk.php [hunk 1]
```diff
@@ -44,8 +44,18 @@ public function getAllPagerFormatted($admin_user_id, $options = [])
             ->with(['favorites', 'comments']);
         $formatted = [];
 
+        if ($where) {
```
Comment (mdwheele) [inline classes/Domain/Entity/Mapper/Talk.php]: After merging, I had to make quite a few modifications to this code to get it to function as I expected it to. Specifically, the incomplete introduction of `$sort` (replacing `$options`) caused several notices that it was not properly initialized and a fatal error on calls to `Talk::getSortOptions`. I have resolved this and think that this PR is salvageable, but would advise that we look at cleaner ways of implementing these 4 export features. Michael has already said this work was pretty-much proof-of-concept to scratch an itch.

**Comment by mdwheele** [comment]
Even with above comment, I think we can merge the work as-is and revisit soon. It does work as expected once I applied a few patches here and there to remedy surface-level issues. It is not easy to write clear tests for this code. Since it is relatively self-contained as a feature (except for fact that it adds to `Talk` mapper method signatures that may need to be cleaned up, it's reasonable for merge.

**Comment by mdwheele** [comment]
@michaelcullum I have taken your work forward at #328. I'm closing this PR as the extension includes all these commits and makes a few improvements on what you already have done. Thanks again for sending this. Awesome stuff!

**Comment by michaelcullum** [comment]
@mdwheele Thanks for taking this forward. As I explained in my initial PR this was more of a POC just to get something working I could use but couldn't really put any more time into it.

**Pull request closed without merging**